doc/source/reference/random/c-api.rst

```diff
@@ -121,57 +135,70 @@ It provides low-level functions for various distributions. All the functions req
+
+.. c:type:: binomial_t
+
+    .. code-block:: c
 
-.. c:function:: int64_t random_negative_binomial(bitgen_t *bitgen_state, double n, double p)
+        typedef struct s_binomial_t {
+          int has_binomial; /* !=0: following parameters initialized for binomial */
+          double psave;
+          RAND_INT_TYPE nsave;
+          double r;
+          double q;
+          double fm;
+          RAND_INT_TYPE m;
+          double p1;
+          double xm;
+          double xl;
+          double xr;
+          double c;
+          double laml;
+          double lamr;
+          double p2;
+          double p3;
+          double p4;
+        } binomial_t;
+     
 
-.. c:function:: int64_t random_binomial(bitgen_t *bitgen_state, double p, int64_t n, binomial_t *binomial)
+.. c:function:: npy_int64 random_binomial(bitgen_t *bitgen_state, double p, npy_int64 n, binomial_t *binomial)
 
-.. c:function:: int64_t random_logseries(bitgen_t *bitgen_state, double p)
+.. c:function:: npy_int64 random_logseries(bitgen_t *bitgen_state, double p)
 
-.. c:function:: int64_t random_geometric_search(bitgen_t *bitgen_state, double p)
+.. c:function:: npy_int64 random_geometric_search(bitgen_t *bitgen_state, double p)
 
-.. c:function:: int64_t random_geometric_inversion(bitgen_t *bitgen_state, double p)
+.. c:function:: npy_int64 random_geometric_inversion(bitgen_t *bitgen_state, double p)
 
-.. c:function:: int64_t random_geometric(bitgen_t *bitgen_state, double p)
+.. c:function:: npy_int64 random_geometric(bitgen_t *bitgen_state, double p)
 
-.. c:function:: int64_t random_zipf(bitgen_t *bitgen_state, double a)
+.. c:function:: npy_int64 random_zipf(bitgen_t *bitgen_state, double a)
 
-.. c:function:: int64_t random_hypergeometric(bitgen_t *bitgen_state, int64_t good, int64_t bad,
-                                    int64_t sample)
+.. c:function:: npy_int64 random_hypergeometric(bitgen_t *bitgen_state, npy_int64 good, npy_int64 bad, npy_int64 sample)
 
-.. c:function:: uint64_t random_interval(bitgen_t *bitgen_state, uint64_t max)
+.. c:function:: npy_uint64 random_interval(bitgen_t *bitgen_state, npy_uint64 max)
 
-.. c:function:: void random_multinomial(bitgen_t *bitgen_state, int64_t n, int64_t *mnix,
-                            double *pix, np.npy_intp d, binomial_t *binomial)
+.. c:function:: void random_multinomial(bitgen_t *bitgen_state, npy_int64 n, npy_int64 *mnix, double *pix, npy_intp d, binomial_t *binomial)
 
-.. c:function:: int random_mvhg_count(bitgen_t *bitgen_state,
-                          int64_t total,
-                          size_t num_colors, int64_t *colors,
-                          int64_t nsample,
-                          size_t num_variates, int64_t *variates)
+.. c:function:: int random_mvhg_count(bitgen_t *bitgen_state, npy_int64 total, size_t num_colors, npy_int64 *colors, npy_int64 nsample, size_t num_variates, npy_int64 *variates)
```

Comment:
I still don't know if `mvhg` is a sensible name, there's no docs other than the function signature and the name is very non-descriptive. Adding a few lines of docs can be addressed after the 1.18.x split, that would be nice (now it's "read the source code" I think). 

For 1.18.x though, can we have a decision on the name? The legacy code has `random_hypergeometric`, so that's taken. The new code could use `hypergeom` or `multi_hypergeom` or something similar perhaps.

Reply:
As far as I can see now, ` random_mvhg_count` and ` random_mvhg_marginals` are the only two poor naming choices that are left, everything else looks clear to me.

Reply:
@WarrenWeckesser any ideas on nicer names?

Reply:
I agree that the names `random_mvhg_count` and `random_mvhg_marginals` are not great for a public-facing API.  The verbose, completely explicit name for the `marginals` version would be

    random_multivariate_hypergeometric_marginals

but I don't think anyone wants a name that long.  Is

    random_multivar_hypergeom_marginals

still too long?

Reply:
@rgommers wrote:
> I still don't know if mvhg is a sensible name, there's no docs other than the function signature

The C functions `random_hypergeometric`,  `random_mvhg_count` and `random_mvhg_marginals` have documentation in the comments of the C code.  These comments explain the arguments, including assumptions about the input values (e.g. input values are expected to be nonnegative) that are not checked in the C code.

Is the plan to eventually expand the documentation in this file (doc/source/reference/random/c-api.rst) to include such information?

Reply:
yes